enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,545 @@
+
+When reconciling a PO custom resource, the PO manager will need to propagate the underlying BundleDeployment's status back to the PO custom resource that's currently being processed, and relaying that individual PO status information back to the "aggregate" ClusterOperator that controller is managing.
+
+### Workflow Description
+
+```mermaid
+sequenceDiagram
+actor Admin
+participant Installer
+participant Bootstrap
+participant CVO
+participant ClusterOperator
+participant PO manager
+Installer->>Admin: Creates the Kubernetes manifests
+CVO->>PO custom resource: Generates
+Admin-->>Installer: Edits Kubernetes manifests
+Installer->>Bootstrap: Creates
+Bootstrap->>CVO: Deploys
+CVO->>PO manager: Deploys
+PO manager->>ClusterOperator: Manages
+PO manager->>PO custom resource: Watches
+CVO->>ClusterOperator: Watches
+```
+
+Note: The above workflow requires admins to manually create, and edit, the cluster Kubernetes manifests with their desired PO intent. There's a chance that admins mistype the package names, and depending on whether upfront validation is possible, that invalid configuration may not be surfaced until after cluster creation. This behavior may require explicit documentation to reduce this risk. These are obvious UX gaps with this workflow, which can be solidified in future phases, but it will result in zero installer changes during the 4.12 timeframe.
+
+### API Extensions
+
+When the "TechPreviewNoUpgrades" feature set is enabled for a cluster, several API extensions will be added during phase 0.
+
+#### OpenShift API(s)
+
+> Note: The following API(s) will live in the openshift/api repository.
+
+- Introduces a top-level PlatformOperator custom-scoped v1alpha1 CRD:
+
+```go
+const (
+    TypeSourced = "Sourced"
+    TypeApplied = "Applied"
+
+    ReasonSourceFailed     = "SourceFailed"
+    ReasonSourceSuccessful = "SourceSuccessful"
+    ReasonApplyFailed      = "ApplyFailed"
+    ReasonApplySuccessful  = "ApplySuccessful"
+)
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:resource:scope=Cluster
+
+// PlatformOperator is the Schema for the platformoperators API.
+type PlatformOperator struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    Spec   PlatformOperatorSpec   `json:"spec,omitempty"`
+    Status PlatformOperatorStatus `json:"status,omitempty"`
+}
+
+// PlatformOperatorSpec defines the desired state of PlatformOperator.
+type PlatformOperatorSpec struct {}
+
+// PlatformOperatorStatus defines the observed state of PlatformOperator.
+type PlatformOperatorStatus struct {
+    // Conditions represents the latest available observations of a platform operator's current state.
+    // +optional
+    // +patchMergeKey=type
+    // +patchStrategy=merge
+    Conditions []metav1.Condition `json:"conditions,omitempty"` patchStrategy:"merge" patchMergeKey:"type"
```

Comment:
I think the backticks on this line need checking, not sure the additional tags work for CRDs as well btw, can we check that

Reply:
I _think_ you're right and the additional tags don't work for CRDs, but I was mainly just copy/pasting what I had seen in o/api when I searched for "language:go conditions". Given this is a tech preview API, I think we can just get away removing the additional markers, and just specifying +optional and omitempty here?

Reply:
(Dropped these in the latest commit :+1:)